crates/holochain/src/conductor/space.rs

```diff
@@ -89,19 +102,21 @@ pub struct TestSpace {
 
 impl Spaces {
     /// Create a new empty set of [`DnaHash`] spaces.
-    pub fn new(
-        root_env_dir: EnvironmentRootPath,
-        db_sync_level: DbSyncStrategy,
-        queue_consumer_map: QueueConsumerMap,
-    ) -> Self {
+    pub fn new(root_env_dir: EnvironmentRootPath, db_sync_strategy: DbSyncStrategy) -> Self {
         Spaces {
             map: RwShare::new(HashMap::new()),
             root_env_dir: Arc::new(root_env_dir),
-            db_sync_level,
-            queue_consumer_map,
+            db_sync_strategy,
+            queue_consumer_map: QueueConsumerMap::new(),
         }
     }
 
+    /// Get something from every space
+    pub fn get_from_spaces<R, F: FnMut(&Space) -> R>(&self, f: F) -> Vec<R> {
```

Comment:
`FnOnce` tends to be more ergonomic to work with from the usage side in cases like this, and the underlying share_ref takes an FnOnce.

```suggestion
    pub fn get_from_spaces<R, F: FnOnce(&Space) -> R>(&self, f: F) -> Vec<R> {
```